From 66125563d10eb6df7ad8f0666c99246ed120ea8d Mon Sep 17 00:00:00 2001
From: felixlionardo <flionardo@blackberry.com>
Date: Thu, 25 Sep 2025 14:41:53 -0400
Subject: [PATCH] fix faccessat on QNX

---
 gio/glocalfile.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/gio/glocalfile.c b/gio/glocalfile.c
index d9903b4e4..c13256faa 100644
--- a/gio/glocalfile.c
+++ b/gio/glocalfile.c
@@ -1273,7 +1273,11 @@ g_local_file_query_exists (GFile        *file,
 {
   GLocalFile *local = G_LOCAL_FILE (file);
 
+#if defined(__QNX__)
+  return faccessat (0, local->filename, F_OK, AT_EACCESS) == 0;
+#else
   return faccessat (0, local->filename, F_OK, AT_EACCESS | AT_SYMLINK_NOFOLLOW) == 0;
+#endif
 }
 #endif
 
-- 
2.34.1

